packages/base/room.gts

```diff
@@ -568,15 +570,45 @@ export class RoomField extends FieldDef {
                 (commandEvent.content.data.eventId ?? event_id),
           ) as ReactionEvent | undefined;
 
+          let commandResultEvent = this.events.find(
+            (e) =>
+              e.type === 'm.room.message' &&
+              e.content.msgtype === 'org.boxel.commandResult' &&
+              e.content['m.relates_to']?.rel_type === 'm.annotation' &&
+              e.content['m.relates_to'].event_id ===
+                commandEvent.content.data.eventId,
+          ) as CommandResultEvent;
+          let serializedResults: LooseSingleCardDocument[] =
+            commandResultEvent?.content?.result;
+          let r: CommandResult | SearchCommandResult | undefined;
+          if (serializedResults) {
+            // TODO: We should move this code into component when room has been changed
+            if (command.name === 'searchCard') {
+              r = new SearchCommandResult({
+                intent: command.arguments.description,
+                cardIds: serializedResults.map((r) => r.data.id),
+              });
+            } else {
```

Comment:
Unfortunately, this is unavoidable. I tried my best to isolate searchCard but this can only be fixed if the room is not a component and I can utilise a service to help with the isolation task.